instrumentation/net/http/otelhttp/internal/semconv/env_test.go

```diff
@@ -132,3 +132,42 @@ func NewTestHTTPClient() HTTPClient {
 		latencyMeasure:       &testInst{},
 	}
 }
+
+func BenchmarkRecordMetrics(b *testing.B) {
+	benchmarks := []struct {
+		name   string
+		server HTTPServer
+	}{
+		{
+			name:   "empty",
+			server: HTTPServer{},
+		},
+		{
+			name:   "nil meter",
+			server: NewHTTPServer(nil),
+		},
+		{
+			name:   "with Meter",
+			server: NewHTTPServer(noop.Meter{}),
+		},
+	}
+
+	for _, bm := range benchmarks {
+		b.Run(bm.name, func(b *testing.B) {
+			req, _ := http.NewRequest("GET", "http://example.com", nil)
+			_ = bm.server.RequestTraceAttrs("stuff", req)
+			_ = bm.server.ResponseTraceAttrs(ResponseTelemetry{StatusCode: 200})
+
+			b.ReportAllocs()
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				bm.server.RecordMetrics(context.Background(), ServerMetricData{
```

Comment:
If you create the context outside the loop, you'll reduce the number of allocations within the benchmarks, making them more accurate.